Lib/aifc.py

```diff
@@ -179,23 +181,26 @@ def _read_string(file):
         dummy = file.read(1)
     return data
 
-_HUGE_VAL = 1.79769313486231e+308 # See <limits.h>
-
 def _read_float(f): # 10 bytes
-    expon = _read_short(f) # 2 bytes
+    data = f.read(10)
+    if len(data) != 10:
+        raise EOFError
+    expon = int.from_bytes(data[:2], 'big', signed=True)
     sign = 1
     if expon < 0:
         sign = -1
         expon = expon + 0x8000
-    himant = _read_ulong(f) # 4 bytes
-    lomant = _read_ulong(f) # 4 bytes
-    if expon == himant == lomant == 0:
+    mant = int.from_bytes(data[2:], 'big')
+    if not expon and not mant:
         f = 0.0
     elif expon == 0x7FFF:
-        f = _HUGE_VAL
+        f = sys.float_info.max
     else:
-        expon = expon - 16383
-        f = (himant * 0x100000000 + lomant) * pow(2.0, expon - 63)
+        expon = expon - 0x3FFF
+        try:
+            f = math.ldexp(mant, expon - 63)
+        except OverflowError:
```

Comment:
As mentioned in the issue discussion, I'd suggest just letting the `OverflowError` propagate. If we encounter a value like this. But in that case, we should probably also raise for an infinity or NaN above.

So if we want to be internally consistent, I guess the choices are:

- (1) raise on overflow, nan and infinity, or
- (2) return `sys.float_info.max` on overflow, nan and infinity

I find it hard to see any justification for option (2) (especially, turning a `nan` into `sys.float_info.max` seems completely nonsensical) other than that that's what the current code does. Backwards compatibility seems unlikely to be a concern here.